client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,28 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final ReaderGroupRetentionConfig retentionConfig;
+
+    public enum ReaderGroupRetentionConfig {
+        NO_CONSUMPTION_BASED_TRUNCATION(false, false),
```

Comment:
i gave the same comment on the other PR -- do we want to make it an enum?
also "truncate at user streamcut" as the enum value is confusion from the end users perspective.. what is a user streamcut?

i suggested that we use two booleans. 
1. `subscribeForRetention`
2. `acknowledge last checkpoint`
and to ensure that `2` is only available to be set when `1` is true, we can use builder methods and hide the setters for individual booleans.

for example:
```
ReaderGroupConfigBuilder {
...
   ReadGroupConfigBuilder subscribe(boolean acknowledgeAtCheckpoints) {
       this.subscribe = true;
       this.acknowledgeAtCheckpoints = acknowledgeAtCheckpoints;
       return this;
   }
}
```

Reply:
I see enum as a superior choice here because of the following reasons:
1. We don't allow users to make invalid choices (ex: subscriber=false, autoTruncate=true) 
2. Indicates to the user that both these flags are linked to the same functionality.
3. User Friendly - rather than asking users to set flags we let them choose a friendly enum name based on their functional requirements the internal implementation details of what flags we use to implement this are hidden from them.

Reply:
Changed Enum names to indicate these are for ConsumtionBasedRetention